src/userService.ts

```diff
@@ -0,0 +1,134 @@
+        return this.usersData;
+    };
+
+    getUserById(id: number): Partial<Omit<User, 'hobbies'>> | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id) as Partial<User> || null;
+            const reternedUser = {...user};
+            if (user) {
+                delete reternedUser.hobbies
+            }
+            return reternedUser
+        }
+        return null
+        
+    };
+
+    getUserHobbies(id: number): Hobbies | null {
+        if (id) {
+            const user = this.usersData.find(user => user.id === id);
+            
+            return user?.hobbies || null
+        }
+        
+        return null
+    };
+
+    createUser(user: Omit<User, 'id'>): User | null {
+        if (user) {
+            const newUser: User = {
+                ...user,
+                id: Date.now()
+            }
+            this.usersData.push(newUser)
+
+            return newUser
+        } 
+
+        return null
+    }
+
+    deleteUser(id: number): User[] | null {
+        if (id) {
+            return this.usersData = this.usersData.filter(user => user.id !== id)
+        }
+
+        return null
+    }
+
+    updateUser(id: number, partialUser: Partial<Omit<User, "id">>): User | null {
+        if (id && partialUser) {
+            let newUser: unknown;
+            this.usersData = this.usersData.map(user => {
+                if (user.id === id) {
+                    newUser = {
+                        ...user,
+                       ...partialUser
+                    }                    
+                    return newUser as User
+                }
+                return user
+            })
+
+            return newUser as User
+        }
+
+        return null
+    }
+
+    addHobbyToUser(id: number, hobby: string): User | unknown {
+        if (id && hobby) {
+            let newUser: unknown;
+
+            this.usersData = this.usersData.map(user => {
+                if (user.id === id && !user.hobbies.includes(hobby)) {
+                    user.hobbies.push(hobby)
+                    newUser = {
+                        ...user,
+                        hobbies: user.hobbies
+                    }
+                    return newUser as User
+                }
+
+                return user
+            })
+
+            return newUser
+        }
+
+        return null
+    }
+
+    deleteHobbyFromUser(id: number, hobby: string): User | unknown {
+        if (id && hobby) {
+            let newUser: unknown;
+
+            this.usersData = this.usersData.map(user => {
+                if (user.id === id) {
+                    newUser = {
+                        ...user,
+                        hobbies: user.hobbies.filter(userHobby => userHobby !== hobby)
+                    }
+                    return newUser as User
+                }
+
+                return user
+            })
+
+            return newUser
+        }
+
+        return null
+    }
```

Comment:
I would suggest to rewrite it like this

```suggestion
    deleteHobbyFromUser(id: number, hobby: string): User | unknown {
        if (id && hobby) {
            const index = this.usersData.findIndex(({id: userId}) => userId === id);

            if(index !== -1) {
                const filteredHobbies = this.usersData[index].hobbies.filter((userHobby) => userHobby !== hobby)
                this.usersData[index].hobbies = filteredHobbies;

                return filteredHobbies;
            }
        }

        return null
    }
```